compiler/rustc_ast_pretty/src/pprust/mod.rs

```diff
@@ -0,0 +1,105 @@
+#[cfg(test)]
+mod tests;
+
+pub mod state;
+pub use state::{print_crate, AnnNode, Comments, PpAnn, PrintState, State};
+
+use rustc_ast as ast;
+use rustc_ast::token::{Nonterminal, Token, TokenKind};
+use rustc_ast::tokenstream::{TokenStream, TokenTree};
+
+pub fn nonterminal_to_string_no_extra_parens(nt: &Nonterminal) -> String {
+    let mut state = State::new();
+    state.insert_extra_parens = false;
+    state.nonterminal_to_string(nt)
```

Comment:
```suggestion
    State { insert_extra_parens: false, ..State::new() }.nonterminal_to_string(nt)
```

Reply:
This would require changing the visibility of several `State` fields. I added a `pub(super)` constructor which uses FRU syntax.